runtime/src/main/java/org/corfudb/security/tls/ReloadableTrustManager.java

```diff
@@ -0,0 +1,156 @@
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This trust manager reloads the trust store upon certificate verification failure
+ *
+ * Created by zjohnny on 9/18/17.
+ */
+@Slf4j
+public class ReloadableTrustManager implements X509TrustManager {
+    private String trustStorePath, trustPasswordPath;
+    private X509TrustManager trustManager;
+
+    //reduce chance of dos attack
+    private long fileChangeTime;
+
+    public ReloadableTrustManager(String trustStorePath, String trustPasswordPath) throws SSLException {
+        this.trustStorePath = trustStorePath;
+        this.trustPasswordPath = trustPasswordPath;
+        fileChangeTime = 0;
+        reloadTrustManager();
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        try {
+            trustManager.checkClientTrusted(chain, authType);
+        } catch (CertificateException cx) {
+            try {
+                if (reloadTrustManager()) {
+                    trustManager.checkClientTrusted(chain, authType);
+                }
+            } catch (SSLException e) {
+                String message = "Issue occurred while rereading trust store.";
+                log.error(message, e);
+                throw new CertificateException(message, e);
+            }
+        }
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        try {
+            trustManager.checkServerTrusted(chain, authType);
+        } catch (CertificateException cx) {
+            try {
+                if (reloadTrustManager()) {
+                    trustManager.checkServerTrusted(chain, authType);
+                }
+            } catch (SSLException e) {
+                String message = "Issue occurred while rereading trust store.";
+                log.error(message, e);
+                throw new CertificateException(message, e);
+            }
+        }
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+        return new X509Certificate[0];
+    }
+
+    private boolean reloadTrustManager() throws SSLException {
+        File trustFile = new File(trustStorePath);
+        if (!trustFile.exists()) {
+            String errorMessage = "Trust store file {" + trustStorePath + "} doesn't exist.";
+            log.error(errorMessage);
+            throw new SSLException(errorMessage);
+        }
+
+        long lastModified = trustFile.lastModified();
+        if (lastModified == fileChangeTime) {
+            return false;
+        }
```

Comment:
I don't see a immediate need for the fileChangeTime.

It does not really save from DoS attacks, they need to be stopped much before this code runs.